### PR TITLE
Hide close channel button if channel is already closed

### LINKED ISSUE #912: Closed channels still have "CLOSE CHANNEL" button
### Description
On the "CHANNELS" screen if you click on `Pending Force Closing` and `Closed` txs a small popup screen shows up and on that screen there's this button `CLOSE CHANNEL` that if you click on it, it takes you to another screen saying you can close those channels again, but those channels are already closed! 

### Screenshot and/or logs if applicable
image

image

### PR DESCRIPTION
**Pull request by wallclockbuilder**
Fixes #912 

Current flow:
screen shot 2019-02-21 at 7 58 48 pm

screen shot 2019-02-21 at 7 59 38 pm
screen shot 2019-02-21 at 7 59 25 pm
screen shot 2019-02-21 at 7 59 14 pm
screen shot 2019-02-21 at 7 58 59 pm

### CODE REVIEW

File: src/view/channel-detail.js [hunk 1]
```diff
@@ -54,9 +54,14 @@ const ChannelDetailView = ({ store, nav }) => (
         <DetailField name="Balance">
           {store.selectedChannel.localBalanceLabel} {store.unitLabel}
         </DetailField>
-        <Button style={styles.deleteBtn} onPress={() => nav.goChannelDelete()}>
-          <ButtonText style={styles.deleteBtnText}>CLOSE CHANNEL</ButtonText>
-        </Button>
+        {store.selectedChannel.statusLabel === 'Closed' ? null : (
```
Comment (valentinewallace) [inline src/view/channel-detail.js]: We also don't want to display the button when the channel is `pending-force-closing` or `pending-closing`, so a better check might be:
```
{!store.selectedChannel.closingTxId ? null : ( ... ) }
```

**Comment by wallclockbuilder** [comment]
Updated to only display Close Channel button if statusLabel contains the keyword 'close' or 'closing'.
closed
pending force closing
pending open
open
waiting close

File: src/view/channel-detail.js (hunk 1 above)
Comment (wallclockbuilder) [reply src/view/channel-detail.js]: closingTxId doen't work. See result.
screen shot 2019-02-27 at 3 26 54 pm
screen shot 2019-02-27 at 2 22 51 pm
screen shot 2019-02-27 at 2 22 43 pm
screen shot 2019-02-27 at 2 22 36 pm

Comment (wallclockbuilder) [reply src/view/channel-detail.js]: Instead just simply matching for the keyword 'close' or 'closing' works reliably.

**Comment by wallclockbuilder** [comment]
Updated view to only display `Close Channel` button if statusLabel contains the keyword 'close' or 'closing':
pending open
open
waiting close
pending force closing
closed

File: mobile/App.js [hunk 2]
```diff
@@ -1,4 +1,12 @@
 import 'node-libs-react-native/globals';
+import { Platform } from 'react-native';
```
[the remark this answers is not in the document]
Comment (wallclockbuilder) [reply mobile/App.js]: Rebase fixed!

File: src/view/channel-detail.js (hunk 1 above)
Comment (wallclockbuilder) [reply src/view/channel-detail.js]: ``` {/Close|Closing/i.test(store.selectedChannel.statusLabel) ? null : (...)}```

**Comment by valentinewallace** [comment]
@wallclockbuilder For some reason there's changes to the `package-lock.json` checked in as well. Could take those out? Otherwise, looks good!

File: src/view/channel-detail.js [hunk 3]
```diff
@@ -54,9 +54,14 @@ const ChannelDetailView = ({ store, nav }) => (
         <DetailField name="Balance">
           {store.selectedChannel.localBalanceLabel} {store.unitLabel}
         </DetailField>
-        <Button style={styles.deleteBtn} onPress={() => nav.goChannelDelete()}>
-          <ButtonText style={styles.deleteBtnText}>CLOSE CHANNEL</ButtonText>
-        </Button>
+        {/Close|Closing/i.test(store.selectedChannel.statusLabel) ? null : (
```
Comment (tanx) [inline src/view/channel-detail.js]: Change to `/close|closing/i.test(store.selectedChannel.status)` as the labels are language dependent and will change once we start to translate the app into different languages.

File: package-lock.json [hunk 4]
```diff
@@ -8060,11 +8077,13 @@
...
           "version": "1.0.2",
-          "bundled": true
+          "bundled": true,
+          "optional": true
         },
         "yallist": {
           "version": "3.0.2",
-          "bundled": true
+          "bundled": true,
+          "optional": true
```
Comment (tanx) [inline package-lock.json]: Please remove the commit that changes the package-lock.json.

**Comment by tanx** [comment]
Closing in favor of #1047

**Pull request closed without merging**